crates/polars-core/src/chunked_array/cast.rs

```diff
@@ -195,6 +198,33 @@ impl ChunkCast for Utf8Chunked {
                     polars_bail!(ComputeError: "expected 'precision' or 'scale' when casting to Decimal")
                 },
             },
+            #[cfg(feature = "dtype-date")]
+            DataType::Date if !validate_is_number(&self.chunks) => {
```

Comment:
wait sorry, are we doing a whole pass over the array here in `validate_is_number`? if so, that feels like it's more computation that would be needed - is there a way to avoid this?